File: app/src/main/java/com/candy/capture/util/TipsUtil.java
package com.candy.capture.util;

import android.content.Context;
import com.google.android.material.snackbar.Snackbar;
import android.view.View;
import android.widget.Toast;

/**
 * Created by zhanghq on 2016/10/30.
 */

public class TipsUtil {

    public static void showToast(Context context, String msg) {
        Toast.makeText(context.getApplicationContext(), msg, Toast.LENGTH_SHORT).show();
    }

    public static void showSnackbar(View view, String msg) {
        Snackbar.make(view, msg, Snackbar.LENGTH_SHORT).show();
    }

    public static void showSnackbar(View view, String msg, String action, View.OnClickListener listener) {
        Snackbar.make(view, msg, Snackbar.LENGTH_LONG)
                .setAction(action, listener)
                .show();
    }
}

File: commonLibrary/src/main/java/com/candy/commonlibrary/utils/TipsUtil.kt
package com.candy.commonlibrary.utils

import android.content.Context
import android.view.View
import android.widget.Toast
import com.google.android.material.snackbar.Snackbar

/**
 * @Description
 * @Author zhanghaiqiang
 * @Date 2020/9/30 16:21
 */
object TipsUtil {
    fun showToast(context: Context?, msg: String?) {
        context?.apply {
            Toast.makeText(applicationContext, msg, Toast.LENGTH_SHORT).show()
        }
    }

    fun showSnackbar(view: View?, msg: String?) {
        Snackbar.make(view!!, msg!!, Snackbar.LENGTH_SHORT).show()
    }

    fun showSnackbar(view: View?, msg: String?, action: String?, listener: View.OnClickListener?) {
        Snackbar.make(view!!, msg!!, Snackbar.LENGTH_LONG)
                .setAction(action, listener)
                .show()
    }
}